backend/dataall/searchproxy/upsert.py

```diff
@@ -1,15 +1,65 @@
 import logging
+import os
+from abc import ABC, abstractmethod
 from datetime import datetime
+from operator import and_
+
+from sqlalchemy.orm import with_expression
+
+from dataall.db import models
+from dataall.searchproxy import connect
 
 log = logging.getLogger(__name__)
 
 
-def upsert(es, index, id, doc):
-    doc['_indexed'] = datetime.now()
-    if es:
-        res = es.index(index=index, id=id, body=doc)
-        log.info(f'doc {doc} for id {id} indexed with response {res}')
-        return True
-    else:
-        log.error(f'ES config is missing doc {doc} for id {id} was not indexed')
-        return False
+class BaseIndexer(ABC):
+    """API to work with OpenSearch"""
+    _INDEX = 'dataall-index'
+    _es = None
+
+    @classmethod
+    def es(cls):
+        """Lazy creation of the OpenSearch connection"""
+        if cls._es is None:
+            cls._es = connect(envname=os.getenv('envname', 'local'))
```

Comment:
I'd consider handling the case where it fails to connect to ES.
Like eg. in search.py: https://github.com/awslabs/aws-dataall/blob/main/backend/dataall/searchproxy/search.py#L11